libraries/job.rb

```diff
@@ -104,6 +119,57 @@ def whyrun_supported?
     end
 
     #
+    # Executes a Jenkins job.
+    #
+    # @raise [JobDoesNotExist]
+    #   if the job does not exist
+    #
+    action(:build) do
+      unless current_resource.exists?
+        fail JobDoesNotExist.new(new_resource.name, :build)
+      end
+
+      if current_resource.enabled?
+        converge_by("Build #{new_resource}") do
+          execution_opts = {}
+          command_args = [
+            'build',
+            escape(new_resource.name),
+          ]
+
+          command_args << '-s' if new_resource.wait_for_completion
+          command_args << '-v' if new_resource.stream_job_output && new_resource.wait_for_completion
+
+          new_resource.parameters.each_pair do |key, value|
+            command_args << "-p #{key}='#{value}'"
+          end
+
+          execution_opts[:live_stream] = STDOUT if new_resource.stream_job_output
+
+          if new_resource.stream_job_output
+            STDOUT.puts(' ')
+            STDOUT.puts(' ')
+            STDOUT.puts('=======================================')
+            STDOUT.puts("= BEGIN #{new_resource.name} JOB OUTPUT")
+            STDOUT.puts('=======================================')
+            STDOUT.puts(' ')
+          end
+
+          executor.execute!(*command_args, execution_opts)
+
+          if new_resource.stream_job_output
+            STDOUT.puts(' ')
+            STDOUT.puts('=======================================')
+            STDOUT.puts('= END JOB OUTPUT')
+            STDOUT.puts('=======================================')
```

Comment:
@danielsdeleo Is there a cleaner way to write to `Chef::Log`'s STDOUT? The code above will stream a Jenkins job's console output in realtime to the Chef log. We'll be using this new `:build` action on the  `jenkins_job` resource to remote trigger Jenkins jobs from Delivery build cookbooks.